arch/arm/src/armv7-m/arm_schedulesigaction.c

```diff
@@ -155,7 +146,20 @@ void up_schedule_sigaction(struct tcb_s *tcb, sig_deliver_t sigdeliver)
                * as the interrupt return context.
                */
 
-              arm_savestate(tcb->xcp.regs);
+              arm_savestate(tcb->xcp.saved_regs);
```

Comment:
Remove, saved_regs is invalid here.

Reply:
#    define arm_savestate(regs)  (regs = (FAR uint32_t *)CURRENT_REGS)